haystack/nodes/base.py

```diff
@@ -173,10 +218,10 @@ def set_config(self, **kwargs):
 
         :param kwargs: all parameters passed to the __init__() of the Component.
         """
-        if not self.pipeline_config:
-            self.pipeline_config = {"params": {}, "type": type(self).__name__}
+        if not self._pipeline_config:
+            self._pipeline_config = {"params": {}, "type": type(self).__name__}
```

Comment:
Good point, the name of this is terrible ihmo :smile: It was simply `self.pipeline_config` before and the aim was to make it private, but I did not feel like renaming it.

I believe this names comes from the fact that this variable stores the configuration coming from the pipeline configuration file. If I had to choose a name though, I'd be even more generic and call it `self._init_params` or something like that, because that's basically what it contains.

What do you think?

Reply:
From my perspective `_init_params` not really reflects its purpose as this is a dict containing the init params and the type of the component. So basically it's the component_definition or component_config we also use in the (actual) pipeline_config without having dependencies resolved. See https://github.com/deepset-ai/haystack/blob/8bfc5791406b591cbcf6bd7f5e071eeee0be4480/haystack/pipelines/base.py#L1173-L1175 and following lines to fill the params by resolving dependency components.
So I'd say `_component_config` reflects all this and would be a great fit to clarify its purpose.

Reply:
I don't agree 100% but I'll take your suggestion and rename it `_component_config` :grin: 

One more question came to my mind though: why do we store the type?? This dictionary is stored in the instance. We can just get the type from the instance itself...

I'll see if removing it breaks anything. If not, count it gone :smile: